src/main/java/net/minecraft/launchwrapper/LaunchClassLoader.java

```diff
@@ -168,22 +168,27 @@ public void registerTransformer(String transformerClassName) {
             }
 
             final byte[] transformedClass = runTransformers(untransformedName, transformedName, getClassBytes(untransformedName));
-            if (DEBUG_SAVE) {
-                saveTransformedClass(transformedClass, transformedName);
-            }
 
-            final CodeSource codeSource = urlConnection == null ? null : new CodeSource(urlConnection.getURL(), signers);
-            final Class<?> clazz = defineClass(transformedName, transformedClass, 0, transformedClass.length, codeSource);
-            cachedClasses.put(transformedName, clazz);
-            return clazz;
+            if (transformedClass != null) {
```

Comment:
Honestly I don't think there is anything wrong with skipping the null check here. If it is null then it throws an exception which is caught and logged below. Thus adding more detail to the stack trace. More details, the better. So I don't see a need for this protection.

Reply:
IMO it *is* worth checking it, as otherwise it will NPE when trying to save the transformed class (`DEBUG_SAVE == true`), and when not debug saving, the exception it does print *points to the classloader itself NPEing*, not that it couldn't find the class source.

There's also zero point going through the transformer list with null bytes, unless you're expecting one of them might provide the data?!